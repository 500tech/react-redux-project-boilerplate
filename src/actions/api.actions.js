// @flow
import { set } from 'lodash/fp';
import type { ApiAction } from 'types/redux.types';

export const apiAction = (action: ApiAction): ApiAction =>
  set('meta.api', true, action);
