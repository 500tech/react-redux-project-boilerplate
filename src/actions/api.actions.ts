import { BaseAction } from 'types/base-redux.types';

export type HttpMethod =
  | 'get'
  | 'post'
  | 'put'
  | 'delete'
  | 'path'
  | 'head'
  | 'options';

type OnSuccess<ResponseBody> = (data: ResponseBody) => BaseAction;
type OnError = (error: any) => void;

export interface ApiAction<ResponseBody> extends BaseAction {
  meta: {
    api: true;
  };
  payload: {
    networkLabel: string;
    method: HttpMethod;
    path: string;
    data?: any;
    baseUrl?: string;
    onSuccess: OnSuccess<ResponseBody> | OnSuccess<ResponseBody>[];
    onError?: OnError | OnError[];
  };
}
