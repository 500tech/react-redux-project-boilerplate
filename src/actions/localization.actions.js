// @flow
// TODO: remove this file if no localization is needed
import type { BaseAction } from 'types/redux.types';

export const SET_LOCALE: string = '[localization] Set Locale';

export type SetLocaleAction = BaseAction<{|
  +locale: string
|}>;
export type LocalizationAction = SetLocaleAction;

export const setLocale = (locale: string): SetLocaleAction => ({
  type: SET_LOCALE,
  payload: {
    locale
  }
});
