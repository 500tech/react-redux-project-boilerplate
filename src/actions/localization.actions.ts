// TODO: remove this file if no localization is needed
import { BaseAction } from 'types/base-redux.types';

export const SET_LOCALE: string = '[localization] Set Locale';

export interface SetLocaleActionPayload {
  locale: string;
}

export type SetLocaleAction = BaseAction<SetLocaleActionPayload>;
export type LocalizationAction = SetLocaleAction;

export const setLocale = (locale: string): SetLocaleAction => ({
  type: SET_LOCALE,
  payload: {
    locale
  }
});
