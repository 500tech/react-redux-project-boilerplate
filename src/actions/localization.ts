// TODO: remove this file if no localization is needed
import { BaseAction } from 'types/base-redux';
import { LocaleTypes } from 'constants/locales';

export const SET_LOCALE: string = '[localization] Set Locale';

export interface SetLocaleAction extends BaseAction {
  payload: {
    locale: string;
  };
}

export const setLocale = (locale: LocaleTypes) => ({
  type: SET_LOCALE,
  payload: {
    locale
  }
});
