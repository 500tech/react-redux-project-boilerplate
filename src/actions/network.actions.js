// @flow
import type { BaseAction } from 'types/redux.types';

export const START_NETWORK: string = '[network] Start';
export const END_NETWORK: string = '[network] End';

export type StartNetworkAction = BaseAction<{|
  +networkLabel: string
|}>;
export type EndNetworkAction = StartNetworkAction;
export type NetworkAction = StartNetworkAction | EndNetworkAction;

export const startNetwork = (networkLabel: string): StartNetworkAction => ({
  type: START_NETWORK,
  payload: {
    networkLabel
  }
});
export const endNetwork = (networkLabel: string): EndNetworkAction => ({
  type: END_NETWORK,
  payload: {
    networkLabel
  }
});
