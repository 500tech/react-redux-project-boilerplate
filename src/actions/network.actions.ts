import { BaseAction } from 'types/base-redux.types';

export const START_NETWORK: string = '[network] Start';
export const END_NETWORK: string = '[network] End';

export interface NetworkAction extends BaseAction {
  payload: {
    networkLabel: string;
  };
}

export const startNetwork = (networkLabel: string): NetworkAction => ({
  type: START_NETWORK,
  payload: {
    networkLabel
  }
});

export const endNetwork = (networkLabel: string): NetworkAction => ({
  type: END_NETWORK,
  payload: {
    networkLabel
  }
});
