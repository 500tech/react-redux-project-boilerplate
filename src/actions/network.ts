import { BaseAction } from 'types/base-redux';

export const START_NETWORK: string = '[network] Start';
export const END_NETWORK: string = '[network] End';

export interface NetworkAction extends BaseAction {
  payload: {
    networkLabel: string;
  };
}

export const startNetwork = (networkLabel: string) => ({
  type: START_NETWORK,
  payload: {
    networkLabel
  }
});

export const endNetwork = (networkLabel: string): NetworkAction => ({
  type: END_NETWORK,
  payload: {
    networkLabel
  }
});
