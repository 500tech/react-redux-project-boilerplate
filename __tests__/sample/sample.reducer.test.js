import sampleReducer from 'sample/sample.reducer';
import { setPosts } from 'sample/sample.actions';

describe('Reducer: sample', () => {
  test('should save posts', () => {
    const posts = [{ id: 1, title: 'first post', body: 'first content' }];
    const action = setPosts(posts);
    const result = sampleReducer(undefined, action);

    expect(result.posts['1']).toEqual(posts[0]);
    expect(result).toMatchSnapshot();
  });

  test('should override posts when there are existing posts', () => {
    const initialState = {
      posts: {
        1: {
          body: 'first content',
          id: 1,
          title: 'first post'
        }
      }
    };
    const posts = [{ id: 2, title: 'second post', body: 'third content' }];
    const action = setPosts(posts);
    const result = sampleReducer(initialState, action);

    expect(result.posts['1']).toBeUndefined();
    expect(result.posts['2'].title).toEqual('second post');

    expect(result).toMatchSnapshot();
  });
});
