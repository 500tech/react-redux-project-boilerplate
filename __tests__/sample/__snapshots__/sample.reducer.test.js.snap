// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: sample should override posts when there are existing posts 1`] = `
Object {
  "posts": Object {
    "2": Object {
      "body": "third content",
      "id": 2,
      "title": "second post",
    },
  },
}
`;

exports[`Reducer: sample should save posts 1`] = `
Object {
  "posts": Object {
    "1": Object {
      "body": "first content",
      "id": 1,
      "title": "first post",
    },
  },
}
`;
